metabonding/src/lib.rs

```diff
@@ -90,17 +102,20 @@ pub trait Metabonding:
         }
     }
 
-    fn is_in_range(&self, value: u64, min: u64, max: u64) -> bool {
+    #[inline]
+    fn is_in_range(&self, value: Week, min: Week, max: Week) -> bool {
         (min..=max).contains(&value)
     }
 
     fn verify_signature(
         &self,
+        caller: &ManagedAddress,
         root_hash: &ManagedHash<Self::Api>,
         user_delegation_amount: &BigUint,
         signature: &Signature<Self::Api>,
     ) {
-        let mut data = root_hash.as_managed_buffer().clone();
+        let mut data = caller.as_managed_buffer().clone();
+        data.append(root_hash.as_managed_buffer());
         data.append(&user_delegation_amount.to_bytes_be_buffer());
 
         let data_len = data.len();
```

Comment:
There's a wrong message one line below. It should be saying wrong data length, not signature.

Reply:
That code was removed in the next PR, so no changes needed here.

Reply:
ok then, let's merge this one and go from there